.github/workflows/scripts/check_clang_tidy.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+CI_SETUP_CMAKE_ARGS=$1
+
+apt install -y clang-tidy
+
+cd taichi
+python3 -m pip install -r requirements_dev.txt
```

Comment:
do we need these packages to run clang-tidy?

Reply:
Yep, I found cmake need some of these packages like GitPython.

Reply:
ok, installing these packages can take quite some time...